src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildWithGlobalJson.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+
+namespace Microsoft.NET.Build.Tests
+{
+    public class GivenThatWeWantToBuildWithGlobalJson : SdkTest
+    {
+        public GivenThatWeWantToBuildWithGlobalJson(ITestOutputHelper log) : base(log)
+        {}
+
+        [Fact]
+        public void It_fails_build_on_failed_sdk_resolution()
+        {
+            var fakePath = "fakePath";
+            TestProject testProject = new TestProject()
+            {
+                Name = "FailedResolution",
+                IsSdkProject = true,
+                TargetFrameworks = "net5.0"
+            };
+            testProject.AdditionalProperties["SdkResolverHonoredGlobalJson"] = "false";
+            testProject.AdditionalProperties["SdkResolverGlobalJsonPath"] = fakePath;
```

Comment:
Instead of creating these properties specifically, could you create a global.json pointing to an invalid version?  Then you could also have to variants of the test, one with `BuildingInsideVisualStudio` true, and one false, and test the different behaviors.

This would be a full framework only test, since the resolver isn't used for .NET Core MSBuild.  And I'm not sure if we hook up the SDKResolver for full framework tests or not, so it might not easily be possible.  But if the SDKResolver already is hooked up, I'd go ahead and try to make the change.

Reply:
It doesn't seem like the SDK resolver it hooked up, I haven't been able to write this type of test.

Reply:
Now I'm remembering more.  There's not a good test hook for adding an SDK resolver.  That prevents us from testing the MSBuild SDK resolver in end-to-end scenarios, and will also prevent us from testing the workload resolver on full framework.

I think as a follow-up, we should try to add a test hook in MSBuild for this.  IE when looking for SDK resolvers, if an MSBUILDADDITIONALSDKRESOLVERSFOLDER environment variable is specified, then we should look there for SDK resolvers in addition to the SdkResolvers folder in the MSBuild bin directory.  If there are folder names with the same name in both paths, the one from the test hook should override the built-in one (so we can test changes to a resolver that ships in MSBuild).

Could you file an MSBuild issue to add the test hook, and and SDK issue to add end-to-end tests for the global.json resolution once that's done?

Reply:
That makes sense, I've filed https://github.com/dotnet/msbuild/issues/5618 and https://github.com/dotnet/sdk/issues/12810. I'll keep these in my backlog of work, I should be able to get to them in the next few weeks.